### PR TITLE
t/t2204-job-info: Add checks to avoid racyness

### PR DESCRIPTION
**Pull request by chu11**
Per issue #2665, this will fix some eventual consistency racyness in `t2204-job-info`.  It won't fix all racyness (such as the one @dongahn saw in #2654), some will be fixed later via issue #2592.

There's some tiny cleanups in here.  Assuming this PR will go in first, I went ahead and cherry-picked some cheapo cleanups from PR #2655 too.

### CODE REVIEW

**Comment by grondo** [comment]
LGTM. Passes all tests under valgrind now.

While you're in here, I also noticed that this test seems incorrect:
```bash
        jobid=$(submit_job) &&
        ! run_timeout 0.2 flux job wait-event $jobid foobar
```

First, best practice is to avoid `!` where possible in tests and use `test_must_fail` or in this case probably `test_expect_code`. O/w, this test succeeds when the timeout expires or `flux-job wait-event` exits with non-zero status, or dies due to a signal.

Second, the test indicates that `flux job wait-event` should "hang" but instead `flux job wait-event` exits immediately with 
```
flux-job: event 'foobar' never received
```

Since the job has reached its final state.

**Comment by chu11** [comment]
> Second, the test indicates that flux job wait-event should "hang" but instead flux job wait-event exits immediately with

Ahh, good catch.  `flux job wait-event` used to hang when a desired event never occurred, but that's no longer the case for the main eventlog and we've reached the end of it.

**Comment by chu11** [comment]
re-pushed with a few extra cleanup fixes:

- adjust tests that assumed a hang would occur, but that style of `wait-event` no longer exists
- use `test_expect code 142` in a few `run_timeout` locations
- replace `!` with `test_must_fail` in a bunch of locations

File: t/t2204-job-info.t [hunk 1]
```diff
@@ -102,6 +102,29 @@ test_expect_success 'load job-exec,sched-simple modules' '
...
+# the job-manager's queue.  To ensure no raciness in tests, we spin
+# until all of the pending jobs have reached SCHED state, running jobs
+# have reached RUN state, and inactive jobs have reached INACTIVE
+# state.
+
+wait_states() {
+        local i=0
+        while ( [ "$(flux job list -A | jq .state | grep 4 | wc -l)" != "4" ] \
+                || [ "$(flux job list -A | jq .state | grep 8 | wc -l)" != "8" ] \
+                || [ "$(flux job list -A | jq .state | grep 32 | wc -l)" != "4" ]) \
```
Comment (grondo) [inline t/t2204-job-info.t]: Would it be better/simpler here to do `$(flux job list --states=sched | wc -l)` `$(flux job list --states=run | wc -l)` and `$(flux job list --states=inactive | wc -l)` to explicitly count the jobs in these states?

Or would `flux job stats` work?

Comment (chu11) [reply t/t2204-job-info.t]: Ahhh, it would be better!  Good catch.

**Comment by chu11** [comment]
re-pushed with suggested fix, squashed since fix was minor

**Comment by grondo** [comment]
Hm, the asan builder failed with an error I haven't seen before, clearly not related to this PR. I'll restart the builder -- and perhaps open an issue on this one.
```
RROR: test_canonical
=====================
AddressSanitizer:DEADLYSIGNAL
=================================================================
==4533==ERROR: AddressSanitizer: SEGV on unknown address 0x7ffd20000001 (pc 0x7fbee4a15785 bp 0x612000000040 sp 0x7fbee056e790 T1)
==4533==The signal is caused by a READ memory access.
    #0 0x7fbee4a15784 in getenv (/lib/x86_64-linux-gnu/libc.so.6+0x42784)
    #1 0x56514fd675e1 in loop_init /usr/src/src/common/libev/ev.c:2928
    #2 0x56514fd6832b in ev_loop_new /usr/src/src/common/libev/ev.c:3159
    #3 0x56514fd70034 in flux_reactor_create /usr/src/src/common/libflux/reactor.c:80
    #4 0x56514fd6d112 in server_thread test/server_thread.c:139
    #5 0x7fbee4fcf6da in start_thread (/lib/x86_64-linux-gnu/libpthread.so.0+0x76da)
    #6 0x7fbee4af488e in __clone (/lib/x86_64-linux-gnu/libc.so.6+0x12188e)
AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV (/lib/x86_64-linux-gnu/libc.so.6+0x42784) in getenv
Thread T1 created by T0 here:
    #0 0x7fbee593fa8f in pthread_create (/usr/lib/gcc/x86_64-linux-gnu/8/libasan.so+0x4aa8f)
    #1 0x56514fd6dd49 in pmi_server_create test/server_thread.c:211
    #2 0x56514fd6b45b in main test/canonical.c:52
    #3 0x7fbee49f4b96 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21b96)
==4533==ABORTING
ERROR: test_canonical.t - missing test plan
ERROR: test_canonical.t - exited with status 1
===XXX ./src/common/libeventlog/test-suite.log 
```

**Pull request merged**